### PR TITLE
PR containing rpc and connection (both, merged --squash)

### PR DESCRIPTION
**Pull request by nicola**
I tried to merge each of them separately, but the previous batch of commits were adding and deleting each others file, so I merge squashed them and they were perfectly fine.

I run it and it passed the test without failing. However deeper review should take place since no question was ever asked

### CODE REVIEW

File: code/model/SubgoalPolicy.cpp [hunk 1]
```diff
@@ -1023,16 +1040,35 @@ bool SubgoalPolicy::LoadPredDictFile (void)
 	return true;
 }
 
+bool SubgoalPolicy::LoadConnections (void) {
+
+  String path = "qa.text_features";
```
Comment (nicola) [inline code/model/SubgoalPolicy.cpp]: this should use config

File: code/libsrc/libantlr3c-3.1.3/antlr3intstream.lo [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# antlr3intstream.lo - a libtool object file
+# Generated by ltmain.sh - GNU libtool 1.5.26 (1.1220.2.493 2008/02/01 16:58:18)
+#
+# Please DO NOT delete this file!
+# It is necessary for linking the library.
+
+# Name of the PIC object.
+pic_object='.libs/antlr3intstream.o'
+
```
Comment (yala) [inline code/libsrc/libantlr3c-3.1.3/antlr3intstream.lo]: Is this part of merging faster-compile?

File: code/model/IR.cpp [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+// 		// TODO: here we should process the data, the comment below can be useful
+// 		// Otherwise have a look at IR.cpp
+// 		p_Callback->OnIRAnswer (*aAnswer);
+// 		delete aAnswer;
+
+// 		// if ('u' == cType)
+// 		// {
+// 		// 	// used for restricted world runs.	
+// 		// 	size_t iOffset = sizeof (size_t) + 1;
+// 		// 	size_t iQuestionId;
```
Comment (yala) [inline code/model/IR.cpp]: Can we remove these massive commented out blocks?

File: code/model/SubgoalPolicy.cpp (hunk 1 above)
Comment (yala) [inline code/model/SubgoalPolicy.cpp]: Yep, i'll patch that and push it here

**Pull request merged**